src/helpers/reverseHighlight.ts

```diff
@@ -0,0 +1,42 @@
+import { Hit } from '../types';
+import { getPropertyByPath, getReversedHighlight } from '../lib/utils';
+import { TAG_REPLACEMENT } from '../lib/escape-highlight';
+import { component } from '../lib/suit';
+
+export type ReverseHighlightOptions = {
+  // @MAJOR string should no longer be allowed to be a path, only array can be a path
+  attribute: string | string[];
+  highlightedTagName?: string;
+  hit: Partial<Hit>;
+  cssClasses?: {
+    highlighted?: string;
+  };
```

Comment:
This makes types easier to compose.

```suggestion
  cssClasses?: Partial<{
    highlighted: string;
  }>;
```